IdentityCore/src/telemetry/request_telemetry/MSIDLastRequestTelemetry.m

```diff
@@ -135,4 +197,51 @@ - (MSIDLastRequestTelemetrySerializedItem *)createSerializedItem
     return [[MSIDLastRequestTelemetrySerializedItem alloc] initWithSchemaVersion:[NSNumber numberWithInteger:self.schemaVersion] defaultFields:defaultFields errorInfo:self.errorsInfo platformFields:nil];
 }
 
+#pragma mark - Private: Save To Disk
+
+- (void)saveTelemetryToDisk
+{
+    dispatch_barrier_sync(self.synchronizationQueue, ^{
+        [NSKeyedArchiver archiveRootObject:self toFile:[self filePathToSavedTelemetry]];
+    });
+}
+
+- (id)initFromDecodedObjectWithSchemaVersion:(NSInteger)schemaVersion silentSuccessfulCount:(NSInteger)silentSuccessfulCount errorsInfo:(NSMutableArray<MSIDRequestTelemetryErrorInfo *>*) errorsInfo
+{
+    self = [super init];
+    if (self)
+    {
+        _schemaVersion = schemaVersion;
+        _silentSuccessfulCount = silentSuccessfulCount;
+        _errorsInfo = errorsInfo;
+        _synchronizationQueue = [self initializeDispatchQueue];
+    }
+    return self;
+}
+
+- (NSString *)filePathToSavedTelemetry
+{
+    NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
+    NSString *documentsDirectoryPath = [paths objectAtIndex:0];
+    NSString *filePath = [documentsDirectoryPath stringByAppendingPathComponent:@"lastRequest"];
```

Comment:
should we design with multiple accounts in mind? Otherwise we might need to re-design it once multi-account support is added so I'd prefer to make sure design is extensible.